src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor.cs

```diff
@@ -117,111 +113,101 @@ private IEnumerable<DisplayedEndpoint> GetEndpoints()
         return ResourceEndpointHelpers.GetEndpoints(Resource, includeInternalUrls: true);
     }
 
-    private IEnumerable<SummaryValue> GetResourceValues()
+    private IEnumerable<ResourcePropertyViewModel> GetResourceProperties(bool ordered)
     {
-        var resolvedKnownProperties = Resource.ResourceType switch
-        {
-            KnownResourceTypes.Project => _projectProperties,
-            KnownResourceTypes.Executable => _executableProperties,
-            KnownResourceTypes.Container => _containerProperties,
-            _ => _resourceProperties
-        };
+        PopulateMissingKnownProperties();
 
-        // This is a left outer join for the SQL fans.
-        // Return the resource properties, with an optional known property.
-        // Order properties by the known property order. Unmatched properties are last.
-        var values = Resource.Properties
-            .Where(p => !p.Value.HasNullValue && !(p.Value.KindCase == Value.KindOneofCase.ListValue && p.Value.ListValue.Values.Count == 0))
-            .GroupJoin(
-                resolvedKnownProperties,
-                p => p.Key,
-                k => k.Key,
-                (p, k) => new SummaryValue { Key = p.Key, Value = p.Value, KnownProperty = k.SingleOrDefault(), Tooltip = GetTooltip(p.Value) })
-            .OrderBy(v => v.KnownProperty != null ? resolvedKnownProperties.IndexOf(v.KnownProperty) : int.MaxValue);
```

Comment:
`ResourcePropertyViewModel` is now a stateful, mutable object. We cannot create copies here. Instead, we reuse the objects for each render, so the UI works correctly without reverting state.